### PR TITLE
feat: Adding new retailer specific scenario

### PR DESCRIPTION
**Pull request by p-marinha**
Adding a new api for a retailer specific scenario.

### CODE REVIEW

File: static/v1/swagger.yaml [hunk 1]
```diff
@@ -66,6 +66,17 @@ paths:
         '200':
           description: OK
 
+  /commissioned-epcs:
```
Comment (p-marinha) [inline static/v1/swagger.yaml:69]: @nrbarry I wasn't too sure about this title for the api. Please feel free to change it if you wish.

File: src/recall-assistant/ift-service.ts [hunk 2]
```diff
@@ -167,10 +185,430 @@ export async function processPromiseList(promiseList: string[], eventType: strin
...
+      // Issue request to the trace API and save in an array of promises
+      promiseList.push(rp(options));
+    }
+  }
+
+  return Promise.all(promiseList).then((response: any) => {
+    const transactionData = [];
+    response.forEach((data) => {
+      const dataObj = JSON.parse(data);
+      if (type === 'po') {
```
Comment (d-soroker) [inline src/recall-assistant/ift-service.ts:488]: `'PO`` ?

File: src/recall-assistant/ift-service.ts [hunk 3]
```diff
@@ -167,10 +185,430 @@ export async function processPromiseList(promiseList: string[], eventType: strin
...
+        transactionData.push(...dataObj.purchase_orders);
+      } else if (type === 'po') {
```
Comment (d-soroker) [inline src/recall-assistant/ift-service.ts:490]: `'DA'` ?

File: src/recall-assistant/ift-service.ts [hunk 4]
```diff
@@ -167,10 +185,430 @@ export async function processPromiseList(promiseList: string[], eventType: strin
...
+        transactionData.push(...dataObj.despatch_advices);
+      } else if (type === 'po') {
```
Comment (d-soroker) [inline src/recall-assistant/ift-service.ts:492]: `'RA'` ?

File: src/recall-assistant/ift-service.ts [hunk 5]
```diff
@@ -167,10 +185,430 @@ export async function processPromiseList(promiseList: string[], eventType: strin
...
+// get LGTIN using epcClass
+export function getLGTIN(epc: any): { gtin: string, lotOrSerialNo: string, valid: boolean } {
+  let epcClass;
+  let subProductNo1;
+  let subProductNo2;
+  let partialGTIN;
+  let gtin;
+  let lot;
+
+  // handle IBM-issued (non-GS1)
```
Comment (d-soroker) [inline src/recall-assistant/ift-service.ts:512]: `IFT-issued`

File: src/recall-assistant/ift-service.ts [hunk 6]
```diff
@@ -2,7 +2,26 @@ import { config } from '../app';
 import * as rp from 'request-promise-native';
```
[the remark this answers is not in the document]
Comment (maxzuo) [reply src/recall-assistant/ift-service.ts:2]: `rp` seems to be the standard, at least it is what the documentation for the package uses.

File: src/recall-assistant/ift-service.ts [hunk 7]
```diff
@@ -41,9 +41,39 @@ export function getTraceRestraintParameters(location_id: string,
...
 
+/**
+ * Splits rp call into pages by constants.PAGE_SIZE
+ * and returns the entire list as one
+ * 
+ * @param options options for rp call
+ * @param traceParameters parameters for call to IFT API
+ * @param pageURI function that produces the URI for each page
+ */
+function paginate_rp(options, traceParameters: any[], pageURI: Function, pageSize: number = constants.PAGE_SIZE): Promise<any> {
```
Comment (d-soroker) [inline src/recall-assistant/ift-service.ts:52]: Very nice.

**Pull request merged**